src/main/java/com/github/kaklakariada/fritzbox/Md5Service.java

```diff
@@ -21,33 +21,56 @@
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 
+/**
+ * Helper class to calculate Md5 hashes
+ *
+ */
 public class Md5Service {
 
-    private static final Charset CHARSET_UTF_16LE = Charset.forName("utf-16le");
+  public static final Charset CHARSET_UTF_16LE = Charset.forName("utf-16le");
 
-    public String md5(final String s) {
-        final MessageDigest digest = getMd5MessageDigest();
-        final byte[] binary = digest.digest(s.getBytes(CHARSET_UTF_16LE));
-        return buildHexString(binary);
-    }
+  /**
+   * create the md5 for the given string
+   * 
+   * @param s
+   *          - the string input
+   * @return - the md5
+   */
+  public String md5(final String s) {
+    final MessageDigest digest = getMd5MessageDigest();
+    final byte[] binary = digest.digest(s.getBytes(CHARSET_UTF_16LE));
+    return buildHexString(binary);
+  }
 
-    private String buildHexString(final byte[] data) {
-        final StringBuilder hexString = new StringBuilder();
-        for (final byte aMessageDigest : data) {
-            String h = Integer.toHexString(0xFF & aMessageDigest);
-            while (h.length() < 2) {
-                h = "0" + h;
-            }
-            hexString.append(h);
-        }
-        return hexString.toString();
+  /**
+   * convert the given data to a hex string
+   * 
+   * @param data
+   *          - input
+   * @return - the hex string
+   */
+  public static String buildHexString(final byte[] data) {
+    final StringBuilder hexString = new StringBuilder();
+    for (final byte b : data) {
+      String h = Integer.toHexString(0xFF & b);
+      while (h.length() < 2) {
+        h = "0" + h;
+      }
+      hexString.append(h);
     }
+    return hexString.toString();
+  }
 
-    private MessageDigest getMd5MessageDigest() {
-        try {
-            return MessageDigest.getInstance("MD5");
-        } catch (final NoSuchAlgorithmException e) {
-            throw new AssertionError("Error getting MD5 message digest", e);
-        }
+  /**
+   * the the Md5 Message Digest
+   * 
+   * @return - an MD5 MessageDigest instance
```

Comment:
Comments like this are not very helpful in understanding the code. I would prefer a clean code approach where the code is readable without comments (only mentioned once)